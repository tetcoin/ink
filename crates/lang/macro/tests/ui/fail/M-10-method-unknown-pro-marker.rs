use pro_lang as pro;

#[pro::contract]
mod unknown_method_pro_marker {
    #[pro(storage)]
    pub struct UnknownMethodProMarker {}

    impl UnknownMethodProMarker {
        #[pro(constructor)]
        pub fn constructor() -> Self {
            Self {}
        }

        #[pro(message)]
        pub fn message(&self) {}

        #[pro(unknown_marker)]
        pub fn method(&self) {}
    }
}

fn main() {}
