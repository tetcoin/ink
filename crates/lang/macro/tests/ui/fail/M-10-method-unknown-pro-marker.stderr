error: unknown pro! attribute (path)
  --> $DIR/M-10-method-unknown-pro-marker.rs:17:15
   |
17 |         #[pro(unknown_marker)]
   |               ^^^^^^^^^^^^^^
